src/validation.cpp

```diff
@@ -2468,6 +2473,18 @@ bool CChainState::DisconnectTip(BlockValidationState& state, DisconnectedBlockTr
         assert(flushed);
     }
     LogPrint(BCLog::BENCH, "- Disconnect block: %.2fms\n", (GetTimeMicros() - nStart) * MILLI);
+
+    {
+        // Prune locks that began at or after the tip should get moved backward so they get a chance to reorg
+        const int max_height_first = pindexDelete->nHeight - 1;
+        for (auto& prune_lock : m_blockman.m_prune_locks) {
```

Comment:
Couldn't this also be skipped when we are not pruning?

Reply:
(see my other comment, same reasoning)